src/server_manager/web_app/ui_components/outline-gcp-create-server-app.ts

```diff
@@ -0,0 +1,341 @@
+                </div>
+              </div>
+              <div class="section-content">
+                <paper-input value="${
+        this.selectedProjectId}" label="Project ID" always-float-label="" maxlength="100" @value-changed="${
+        this.onProjectIdChanged}"></paper-input>
+              </div>
+            </div>
+            
+            <div class="section">
+              <div class="section-header">
+                <span class="stepcircle">2</span>
+                <div class="instructions">
+                  Choose your preferred billing method for this project
+                </div>
+              </div>
+              <div class="section-content">
+                <paper-dropdown-menu no-label-float="">
+                  <paper-listbox slot="dropdown-content" selected="${
+        this.selectedBillingAccountId}" attr-for-selected="name" @selected-changed="${
+        this.onBillingAccountSelected}">
+                  ${this.billingAccounts.map(billingAccount => {
+      return html`<paper-item name="${billingAccount.id}">${billingAccount.name}</paper-item>`;
+    })}
+                  </paper-listbox>
+                </paper-dropdown-menu>
+              </div>
+            </div>
+        </outline-step-view>
+
+        <outline-region-picker-step id="regionPicker" .localize=${
+        this.localize} @region-selected="${this.onRegionSelected}">  
+        </outline-region-picker-step>
+      </iron-pages>`;
+  }
+
+  private async handleBillingVerificationNextTap(): Promise<void> {
+    this.billingAccounts = await this.account.listBillingAccounts();
+    if (!this.billingAccounts || this.billingAccounts.length === 0) {
+      // TODO: Show error
+    } else {
+      this.showProjectSetup();
+    }
+  }
+
+  private isProjectSetupNextEnabled(projectId: string, billingAccountId: string): boolean {
+    // TODO: Proper validation
+    return projectId !== '' && billingAccountId !== '';
+  }
+
+  private async handleProjectSetupNextTap(): Promise<void> {
+    this.project =
+        await this.account.createProject(this.selectedProjectId, this.selectedBillingAccountId);
+    this.showRegionPicker();
+  }
+
+  private showBillingAccountSetup(): void {
+    this.currentPage = 'billingAccountSetup';
+  }
+
+  private showProjectSetup(): void {
+    this.currentPage = 'projectSetup';
+  }
+
+  private async showRegionPicker(): Promise<void> {
+    this.currentPage = 'regionPicker';
+    const regionMap = await this.account.listLocations(this.project.id);
+    const locations = Object.entries(regionMap).map(([regionId, zoneIds]) => {
+      return this.createLocationModel(regionId, zoneIds);
+    });
+    this.regionPicker.locations = locations;
+  }
+
+  async start(account: GcpAccount): Promise<void> {
+    this.init();
+    this.account = account;
+
+    const projects = await this.account.listProjects();
+    if (projects && projects.length > 0) {
+      this.project = projects[0];
```

Comment:
You need to check if the project is in the right state. Perhaps something like `this.account.isProjectSetup()`;
That should check the project exists, billing is setup and services are enabled.